daemon/mgr/image.go

```diff
@@ -89,3 +119,87 @@ func (mgr *ImageManager) SearchImages(ctx context.Context, name string, registry
 
 	return nil, nil
 }
+
+// GetImage gets image by image id or ref.
+func (mgr *ImageManager) GetImage(ctx context.Context, idOrRef string) (*types.Image, error) {
+	return mgr.cache.get(idOrRef)
+}
+
+func (mgr *ImageManager) loadImages() error {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	images, err := mgr.client.ListImages(ctx)
+	cancel()
+	if err != nil {
+		return err
+	}
+
+	for i := range images {
+		mgr.cache.put(&images[i])
+	}
+	return nil
+}
+
+type imageCache struct {
+	sync.Mutex
+	refs map[string]*types.Image // store the mapping ref to image.
+	ids  *patricia.Trie          // store the mapping id to image.
+}
+
+func newImageCache() *imageCache {
+	return &imageCache{
+		refs: make(map[string]*types.Image),
+		ids:  patricia.NewTrie(),
+	}
+}
+
+func (c *imageCache) put(image *types.Image) {
+	c.Lock()
+	defer c.Unlock()
+
+	id := strings.TrimPrefix(image.Digest, "sha256:")
+	ref := image.Name
+
+	c.refs[ref] = image
+	c.ids.Insert(patricia.Prefix(id), image)
+}
+
+func (c *imageCache) get(idOrRef string) (*types.Image, error) {
+	c.Lock()
+	defer c.Unlock()
+
+	image, ok := c.refs[idOrRef]
+	if ok {
+		return image, nil
+	}
+
+	var images []*types.Image
+
+	fn := func(prefix patricia.Prefix, item patricia.Item) error {
+		if image, ok := item.(*types.Image); ok {
+			images = append(images, image)
+		}
+		return nil
+	}
+
+	if err := c.ids.VisitSubtree(patricia.Prefix(idOrRef), fn); err != nil {
+		// the error does not occur.
+		return nil, err
+	}
+
+	if len(images) != 1 {
+		return nil, fmt.Errorf("found %d images, not only one: %s", len(images), idOrRef)
```

Comment:
It would lead to the failure of pound create testcases.
Even if image is not exist, "pouch create" still return 0. 
ping @Letty5411 , could you please verify the busybox:latest image is pull before executing pouch create busybox:latest?

Reply:
Thanks @sunyuan3 @Letty5411 
Maybe an issue needs filed to track this.